pkg/soc/soc.go

```diff
@@ -232,3 +235,84 @@ func recoverAddress(signature, digest []byte) ([]byte, error) {
 	}
 	return recoveredEthereumAddress, nil
 }
+
+// CreateSelfAddressedEnvelope create a SOC chunk without the payload (called envelope) addressed to its neighbourhood
+func CreateSelfAddressedEnvelope(ctx context.Context, owner *Owner, overlay swarm.Address, targetLen int, payloadId swarm.Address, signer crypto.Signer) (swarm.Chunk, error) {
+	f := func(id []byte) (swarm.Chunk, error) {
+		newAddr, err := CreateAddress(id, owner)
+		if err != nil {
+			return nil, err
+		}
+
+		if !bytes.Equal(overlay.Bytes()[:targetLen], newAddr.Bytes()[:targetLen]) {
+			return nil, nil
+		}
+
+		// build the soc envelope
+		data := make([]byte, swarm.SpanSize+swarm.HashSize)
+		binary.BigEndian.PutUint64(data[:swarm.SpanSize], swarm.HashSize)
+		copy(data[swarm.SpanSize:], payloadId.Bytes())
+		chunk := swarm.NewChunk(payloadId, data)
+		s := New(id, chunk)
+		err = s.AddSigner(signer)
+		if err != nil {
+			return nil, err
+		}
+		return s.ToChunk()
+	}
+	return mine(ctx, f)
+}
+
+// mine finds out the nonce that is valid to create a envelope address
+func mine(ctx context.Context, f func(nonce []byte) (swarm.Chunk, error)) (swarm.Chunk, error) {
```

Comment:
This function looks the same as https://github.com/ethersphere/bee/pull/695/files#diff-3fc846f43f3c15fe3d5e0d8f0ad7b4bcR217 in a different PR. Consider reusing the same function when both PRs get merged, instead copying it.

Reply:
whichever comes first :-)

Reply:
janos left a comment here about  not copying..........

Reply:
pointed this to tojan.Mine